### ISSUE TITLE
This is Virus !

### ISSUE THREAD

**Issue by Chamiu** [cazala/coin-hive#5]
訪問者に事前通知なく、訪問者のCPUを無断で使用するcoin-hive.comは、明白なウイルスの流布者だ。
恥を知れ。
Coin-hive.com, which uses visitor's CPU without notice to visitors without permission, is an explicit virus spreader.
Shame on you.

**Comment by frankkienl** [comment]
As far as I know, they ask permission. You have to click a start button before they start using the CPU. Therefor, it's **not a virus**. 

As website using coin-hive _could_ start the miner without prior permission. But that would make that website a scammer. Blame those websites, not the technology.

**Comment by cazala** [comment]
But this doesn't run it on your visitors website.. This allows you to run Coin-Hive's client from node.js, so that means you need a server to run it. So you would be _stealing_ your own server's CPU in that case..

**Comment by Chamiu** [comment]
Hi, CPU stealer [coin-hive.com](http://archive.is/XeZaO).

```
<script src="https://coin-hive.com/lib/coinhive.min.js"></script>
<script>var miner = new CoinHive.Anonymous('hjouJXuqZk51vx159qgzz47eJagKJvzR', { throttle: 0.2});miner.start();</script>
```

Where permission ?

**Comment by Chamiu** [comment]
Maintain erased writes.

```
As far as I know, they ask permission. You have to click a start button before they start using the CPU. Therefor, it's not a virus.

As website using coin-hive could start the miner without prior permission. But that would make that website a scammer. Blame those websites, not the technology.
```

https://github.com/cazala/coin-hive/issues/5#issuecomment-330152430

Where permission ?

**Comment by soyuka** [comment]
Dude @cazala  has nothing to do with coin-hive according to readme. Please complain to them instead?

**Comment by rained23** [comment]
why is this a virus ? since when running javascript in user CPU considered virus ?

**Comment by Chamiu** [comment]
very interesting.
This repository name is "coin-hive"
CPU stealer domain is coin-hive.com.

**Comment by soyuka** [comment]
> why is this a virus ? since when running javascript in user CPU considered virus ?

Because if you click on the "start" button it'll start mining coins from the website and use your CPU power to gain them coins. Nothing to do with a virus.

**Comment by Chamiu** [comment]
coin-hive.com hide information at whois.
> Dude @cazala has nothing to do with coin-hive.
It's proof of the devil.

Why not is virus ? without notice to visitors without permission.

> if you click on the "start" button it'll start mining coins from the website

https://github.com/cazala/coin-hive/issues/5#issuecomment-330152430
WHERE is "START BUTTON" ? 
WHERE is "NOTICE" and "PERMISSION" ?

Please explicitly indicate where the code is.

**Comment by rained23** [comment]
Since when you need permission to run Javascript?  Your morale view
shouldn't be imposed to a development issue tracker.

Mining coin while user visiting is another way of monetization the same
like how you show ads in user browser you doesn't need to ask user
permission to serve ads.

**Comment by soyuka** [comment]
See: https://github.com/cazala/coin-hive#disclaimer 

![guy-is-blind](https://user-images.githubusercontent.com/1321971/30536199-39699c76-9c65-11e7-9343-3d28d6236c8b.png)

**Comment by SkinyMonkey** [comment]
No. This is a way to fund a website without a need for premium payment/donation etc.
There is two way to use it:
- telling your user what you do and why.
- go stealth and use their CPU without a warning.

My website will do the first one and I don't see any shame in that case.

Extract from coin-hive website:

"While it's possible to run the miner without informing your users, we strongly advise against it. You know this. Long term goodwill of your users is much more important than any short term profits."

Next time read before accusing right away.

**Comment by cazala** [comment]
If what Coin-Hive does is morale or not is out of the scope of this project (although I think some flash ads steal more CPU cycles and are more annoying than their miner). And also I think there's a misunderstanding between coin-hive.com and this project. This is an npm package that only allows you to spawn a headless chrome and run the miner in there, so there's no 'visitors' or users affected other than yourself or your own node.js server.. I'm moving the disclaimer to the top of the readme to see if that helps
